src/NServiceBus.AcceptanceTests/Core/Routing/When_setting_operation_state_in_pipeline.cs

```diff
@@ -47,7 +47,10 @@ public class Context : ScenarioContext
         public class SenderReusingSendOptions : EndpointConfigurationBuilder
         {
             public SenderReusingSendOptions() => EndpointSetup<DefaultServer>((c, r) =>
-                c.Pipeline.Register(new OperationContextModifyingBehavior((Context)r.ScenarioContext), "modifies message operations context values"));
+            {
+                c.Pipeline.Register(new OperationContextModifyingBehavior((Context)r.ScenarioContext), "modifies message operations context values");
+                c.LimitMessageProcessingConcurrencyTo(1);
```

Comment:
what kind of problems does concurrency cause? should we use `Interlocked.Increment` on line 63 instead or are there other issues?